### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by okashansky-mdsol**
This PR contains two GHA workflows:
- `ci`, which runs lint/test/build on push;
- `build_artifact`, which builds new binary on new tags/releases (it is expected that the artifact will be attached to the release.

### CODE REVIEW

**Comment by okashansky-mdsol** [comment]
@jcarres-mdsol @ykitamura-mdsol, please review.

File: .github/workflows/build_artifact.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  push:
+    tags:
+      - v*
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: [ ubuntu-latest ]
```
Comment (ykitamura-mdsol) [inline .github/workflows/build_artifact.yml:14]: not running on `self-hosted`?

```suggestion
    runs-on: [ self-hosted, ubuntu-latest ]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - go.sum
+      - Makefile
+      - register.go
+
+permissions:
+  contents: read
+
+jobs:
+  verify:
+    runs-on: [ ubuntu-latest ]
```
Comment (ykitamura-mdsol) [inline .github/workflows/ci.yml:18]:
```suggestion
    runs-on: [ self-hosted, ubuntu-latest ]
```

Comment (okashansky-mdsol) [reply .github/workflows/ci.yml:18]: I intentionally did it. The repository is public, so sooner or later, third-party participants' code can be run here. I don't think their GitHub Actions should be allowed to run in our cloud.

File: .github/workflows/build_artifact.yml (hunk 1 above)
Comment (okashansky-mdsol) [reply .github/workflows/build_artifact.yml:14]: please see my answer below

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ykitamura-mdsol) [reply .github/workflows/ci.yml:18]: Oh, public! 👍

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+          go-version: '1.22'
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.58.2
+          args: --verbose --concurrency=2 --timeout=600s
+
+      - name: Test with the Go CLI
+        run: go test -v ./...
```
Comment (ykitamura-mdsol) [inline .github/workflows/ci.yml:42]: there are no test files (yet) but I'm sure you have a plan to add them 😉

**Review by ykitamura-mdsol — APPROVED** [review]

**Pull request merged**